src/mlio-py/mlio/core/data_reader.cxx

```diff
@@ -280,46 +273,11 @@ register_data_readers(py::module &m)
              })
         .def("__next__", &py_data_iterator::next);
 
-    py::class_<data_reader, py_data_reader, intrusive_ptr<data_reader>>(
+    py::class_<data_reader_params, std::shared_ptr<data_reader_params>>(
```

Comment:
I know that it is going to be mostly boilerplate code, but in addition to constructor parameters, I also prefer to have read/write accessor properties.

Reply:
Sounds good, I'll add them.